tests/components/group/test_fan.py

```diff
@@ -350,6 +360,110 @@ async def test_direction_oscillating(hass, setup_comp):
+            ],
+        },
+    )
+    hass.states.async_set(
+        PERCENTAGE_FULL_FAN_ENTITY_ID,
+        STATE_ON,
+        {
+            ATTR_SUPPORTED_FEATURES: FULL_SUPPORT_FEATURES,
+            ATTR_PRESET_MODE: PRESET_MODE_SMART,
+            ATTR_PRESET_MODES: [
+                PRESET_MODE_AUTO,
+                PRESET_MODE_ON,
+                PRESET_MODE_SLEEP,
+                PRESET_MODE_SMART,
+            ],
+        },
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(FAN_GROUP)
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_FRIENDLY_NAME] == DEFAULT_NAME
+    assert state.attributes[ATTR_ENTITY_ID] == [*FULL_FAN_ENTITY_IDS]
+    assert ATTR_ASSUMED_STATE not in state.attributes
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == FULL_SUPPORT_FEATURES
+    assert state.attributes[ATTR_PRESET_MODE] == PRESET_MODE_SMART
+    assert ATTR_ASSUMED_STATE not in state.attributes
+
+    # Test assumed state
+    # Add Entity with a different preset_mode should set assumed state
+    hass.states.async_set(
+        PERCENTAGE_FULL_FAN_ENTITY_ID,
+        STATE_ON,
+        {
+            ATTR_SUPPORTED_FEATURES: FULL_SUPPORT_FEATURES,
+            ATTR_PRESET_MODE: PRESET_MODE_ON,
+            ATTR_PRESET_MODES: [
+                PRESET_MODE_AUTO,
+                PRESET_MODE_ON,
+                PRESET_MODE_SLEEP,
+                PRESET_MODE_SMART,
+            ],
+        },
+    )
+
+    await hass.async_block_till_done()
+    state = hass.states.get(FAN_GROUP)
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_ASSUMED_STATE] is True
+    assert ATTR_ASSUMED_STATE in state.attributes
+    assert state.attributes[ATTR_PRESET_MODE] is None
+
+    # Now that everything is the same, no longer assumed state
+    # Group preset_mode only when all members has same preset_mode
+    hass.states.async_set(
+        LIVING_ROOM_FAN_ENTITY_ID,
+        STATE_ON,
+        {
+            ATTR_SUPPORTED_FEATURES: FULL_SUPPORT_FEATURES,
+            ATTR_PRESET_MODE: PRESET_MODE_SLEEP,
+            ATTR_PRESET_MODES: [
+                PRESET_MODE_AUTO,
+                PRESET_MODE_ON,
+                PRESET_MODE_SLEEP,
+                PRESET_MODE_SMART,
+            ],
+        },
+    )
+    hass.states.async_set(
+        PERCENTAGE_FULL_FAN_ENTITY_ID,
+        STATE_ON,
+        {
+            ATTR_SUPPORTED_FEATURES: FULL_SUPPORT_FEATURES,
+            ATTR_PRESET_MODE: PRESET_MODE_SLEEP,
+            ATTR_PRESET_MODES: [
+                PRESET_MODE_AUTO,
+                PRESET_MODE_ON,
+                PRESET_MODE_SLEEP,
+                PRESET_MODE_SMART,
+            ],
+        },
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get(FAN_GROUP)
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_PRESET_MODE] == PRESET_MODE_SLEEP
+    assert ATTR_ASSUMED_STATE not in state.attributes
+
```

Comment:
Please add tests for fans that are group together that

- have completely different preset modes 
- have only one overlapping preset mode

Reply:
Tests added.